### ISSUE TITLE
Dokka 0.10.0 ignore unsuppresed packages in suppressed packages

### ISSUE THREAD

**Issue by fAntel** [Kotlin/dokka#521]
**Describe the bug**
Dokka 0.10.0 doesn't generate documentation for "supprese = false" packages located in "suppress = true" package.

**Expected behaviour**
In Dokka 0.9.18 documentation for such packages was generated.

**Screenshots**
[0.9.18](https://user-images.githubusercontent.com/10121347/68030043-a77b1c00-fcc9-11e9-81c6-720cc7389c5e.png)
[0.10.0](https://user-images.githubusercontent.com/10121347/68030056-aea22a00-fcc9-11e9-810e-e4fea66474f5.png)

**To Reproduce**
Run ./gradlew dokka on this project and check the results. There are configuration for 0.10.0 and for 0.9.18 (commented)
[example.zip](https://github.com/Kotlin/dokka/files/3798222/example.zip)

**Dokka configuration**
Configuration of dokka used to reproduce the bug

```groovy
dokka { 
    outputFormat = 'html'
    configuration {
        perPackageOption {
            prefix = <some package>
            suppress = true
        }

        perPackageOption {
            prefix = <package inside previuos package>
            suppress = false
        }
    }
}
```

**Installation**
- Operating system: Windows
- Build tool: Gradle v5.4.1
- Dokka version: 0.10.0

**Comment by kamilok1965** [comment]
Hi @fAntel !
So there's an easy fix for that, you just need to reverse the `perPackageOption`s order, to have the more specific first and the least specific last. 
The configuration shouldn't depend on the order of declarations though, so it will be fixed

**Comment by fAntel** [comment]
> Hi @fAntel !
> So there's an easy fix for that, you just need to reverse the `perPackageOption`s order, to have the more specific first and the least specific last.
> The configuration shouldn't depend on the order of declarations though, so it will be fixed

For attached example (and for my project) reversing order of `prePackageOption`s don't help. But I'll wait for next version of dokka and check. I agree that it shouldn't depend on the order of declarations.

**Comment by kamilok1965** [comment]
Fixed in d59f77a81fcbc94a074e7b481af299765342a7a2

**Comment by martinbonnin** [comment]
I've been working on adding regex support to `perPackageOption` (https://github.com/Kotlin/dokka/pull/598/files).

It's hard to tell based on a regex which one is more specific so it'll be hard apply the same logic and make the end-result independent of the order of declarations so for regexes, the first match will be taken.
